### PR TITLE
Add minimum and maximum photos per row constraints in rows layout

### PR DESCRIPTION
**Pull request by ayushnirwal**
- [x] I have followed the 'Sending a Pull Request' section of the [contributing guide](https://github.com/igordanchenko/react-photo-album/blob/main/CONTRIBUTING.md#sending-a-pull-request)

### Summary 
This PR introduces 2 new props which provide more control to the end-user on row layout.

`minPhotoCount` - absolute minimum number of elements (Photos) allowed in a row.
`maxPhotoCount` - targeted maximum number of elements (Photos) allowed in a row. 

### Techincal choices
-  `maxPhotoCount` will be overridden by `limitNodeSearch` if its value is greater than the guestimated value for performance's sake.

### CODE REVIEW

**Comment by igordanchenko** [comment]
Hi there! Can you describe the issue you are trying to address with this PR? I'd like to understand your use case. Some examples (screenshots or code sandbox) will be very helpful.

**Comment by ayushnirwal** [comment]
> Hi there! Can you describe the issue you are trying to address with this PR? I'd like to understand your use case. Some examples (screenshots or code sandbox) will be very helpful.

Hello!
You can see the use case and screenshots in discussions of this [PR](https://github.com/GoogleForCreators/web-stories-wp/pull/10463).
TL;DR of that conversation would be that just using `targetedRowHeight` is not giving us desired results. So implementing a limit ( minimum and maximum ) would give us more direct control over the end result.

**Comment by igordanchenko** [comment]
Thank you! 
I'm going to pull min/max props one level down and introduce just one top-level prop `rowConstraints` since `minPhotos`/`maxPhotos` props are applicable only to `rows` layout. 
Please also keep in mind that it won't be a good idea to set both `minPhotos` and `maxPhotos` to 2 unless you are planning to display only even number of photos. So what you'd probably want to do is to set `maxPhotos` to 2, and decrease the `targetRowHeight` value.

**Comment by igordanchenko** [comment]
@ayushnirwal ,

I also couldn't help myself but notice a potential bug in the way you are using spacing/margin props in [10463](https://github.com/GoogleForCreators/web-stories-wp/pull/10463). PhotoAlbum's `spacing` prop defines spacing between photos (think in terms of grid gap). `layout.width` and `layout.height` provide the exact calculated dimensions of an image, so you don't want to subtract `2 * margin` here as otherwise you'd break your images' aspect ratios and your images would look stretched.

So instead of

```
margin={PHOTO_MARGIN + 'px'}
width={layout.width - 2 * PHOTO_MARGIN}
height={layout.height}
```

you'd probably want to do something along these lines

```
marginBottom={PHOTO_MARGIN + 'px'}
width={layout.width}
height={layout.height}
```

You need to take care of the vertical spacing yourself since you removed row wrapper divs, and flexbox will take care of the horizontal spacing (I'd give `justify-content: space-between` a shot here)

**Comment by igordanchenko** [comment]
Oh, and also PhotoAlbum is exported as both default and named exports so you can import it either as

```
import { PhotoAlbum as Gallery } from 'react-photo-album';
```

or 

```
import Gallery from 'react-photo-album';
```
